packages/backend/src/shared/pagination.ts

```diff
@@ -0,0 +1,80 @@
+import { BaseModel, PageInfo, Pagination } from './baseModel'
+
+export function parsePaginationQueryParameters(
+  first?: string | string[],
+  last?: string | string[],
+  cursor?: string | string[]
+): Pagination {
+  if (
+    (first !== undefined && typeof first !== 'string') ||
+    (last !== undefined && typeof last !== 'string') ||
+    (cursor !== undefined && typeof cursor !== 'string')
+  ) {
+    throw new Error('Query parameters are not strings')
+  }
+  if (first && last) {
+    throw new Error('first and last provided. Only one allowed')
+  }
+  if (last && !cursor) {
+    throw new Error('cursor needed for backwards pagination')
+  }
+  return {
+    first: Number(first) || undefined,
+    last: Number(last) || undefined,
+    before: last ? cursor : undefined,
+    after: cursor && !last ? cursor : undefined
+  }
+}
+
+export async function getListPageInfo<T extends BaseModel>(
+  getPage: (pagination: Pagination) => Promise<T[]>,
+  page: T[],
+  pagination?: Pagination
+): Promise<PageInfo> {
+  const pageInfo = await getPageInfo((pagination) => getPage(pagination), page)
+  if (pagination?.last) {
+    pageInfo.last = page.length
+  } else {
+    pageInfo.first = page.length
+  }
+  return pageInfo
+}
+
+export async function getPageInfo<T extends BaseModel>(
```

Comment:
What if [`getPage`](https://github.com/interledger/rafiki/blob/557fe12851bb7d34911dcba13d8724cee0403f83/packages/backend/src/shared/baseModel.ts#L41) included `pageInfo` in the query response (similar to [`page`](https://vincit.github.io/objection.js/api/query-builder/other-methods.html#page))?

Reply:
Since `pageInfo` is always part of these list request, we could move it. However, I'd like to do that in a subsequent PR to get the list endpoints in. We can always refactor later.